R/data-storage-log-file.R

```diff
@@ -0,0 +1,185 @@
+  # Public
+  public = list(
+    # @title initialize RSQLite storage provider
+    # @param username string with username of the current session
+    #
+
+    #' @description
+    #' Initialize the data storage class
+    #' @param username string with username of the current session
+    #' @param session_id string with custom session id (should not be used)
+    #' @param log_file_path string with path to JSON log file user actions
+    #' @param session_file_path string with path to JSON log file for the session details
+
+    initialize = function(
+    username, session_id = NULL, log_file_path, session_file_path
+    ) {
+      super$initialize(username, session_id)
+      logger::log_info("path to file: {log_file_path}")
+      private$connect(log_file_path = log_file_path, session_file_path = session_file_path)
+    },
+
+    #' @description Insert new data
+    #' @param values list of values to write to database
+    #' @param bucket path to log file; defaults to `log_file_path` used when initialized
+    #' @param add_username boolean flag that indicates if line should include
+    #' the username of the current session
+
+    insert = function(values, bucket = private$log_file_path, add_username = TRUE) {
+      checkmate::assert_list(values)
+      checkmate::assert_string(bucket)
+      checkmate::assert_logical(add_username)
+
+      values <- private$insert_checks(values, add_username = add_username)
+
+      private$write(values, bucket = bucket)
+    },
+
+    #' @description read all user data from SQLite
+    #' @param date_from date representing the starting day of results
+    #' @param date_to date representing the last day of results
+
+    read_user_data = function(date_from, date_to) {
+      log_data <- private$read_data(private$log_file_path, date_from, date_to)
+
+      if (NROW(log_data) > 0) {
+        return(dplyr::mutate(log_data, date = as.Date(.data$time)))
+      }
+      log_data
+    },
+
+    #' @description read all session data from SQLite
+    #' @param date_from date representing the starting day of results
+    #' @param date_to date representing the last day of results
+
+    read_session_data = function(date_from, date_to) {
+      db_data <- private$read_data(private$session_file_path, date_from, date_to)
+
+      db_data %>%
+        dplyr::select("session", "detail") %>%
+        dplyr::group_by(.data$session) %>%
+        dplyr::summarise(title = paste(.data$detail, collapse = " | "))
+    },
+
+    #' @description read all session data
+
+    close = function() {
+      private$close_connection()
+    }
+  ),
+  active = list(
+
+    #' @field action_bucket string that identifies the file path to store user
+    #' related and action data
+
+    action_bucket = function() {
+      private$log_file_path
+    },
+
+    #' @field session_bucket string that identifies the bucket to store session
+    #' details data
+
+    session_bucket = function() {
+      private$session_file_path
+    }
+  ),
```

Comment:
Users can now access the file paths they specified when initializing through these active bindings. It also makes it consistent when switching data storage and using the same active bindings in other functions, e.g. `log_session_detail()`.